### PR TITLE
Feat: Unstake proxy

### PR DESCRIPTION
**Pull request by tuul-wq**
- Unstake proxy
- Update Sign / Submit to accept and handle arrays of transactions

image

part of #1444

### CODE REVIEW

**Review by pgolovkin — APPROVED** [review]

File: src/renderer/entities/transaction/lib/transactionBuilder.ts [hunk 1]
```diff
@@ -43,3 +46,58 @@ function buildTransfer({ chain, accountId, destination, asset, amount, xcmData }
     },
   };
 }
+
+type UnstakeParams = {
+  chain: Chain;
+  asset: Asset;
+  accountId: AccountId;
+  amount: string;
+  chill?: boolean;
```
Comment (Asmadek) [inline src/renderer/entities/transaction/lib/transactionBuilder.ts]: what do you think about `withChill`?

File: src/renderer/widgets/Unstake/model/form-model.ts [hunk 2]
```diff
@@ -0,0 +1,690 @@
...
+
+const $selectedSignatories = createStore<Account[]>([]);
+
+const $unstakeForm = createForm<FormParams>({
+  fields: {
+    shards: {
+      init: [] as Account[],
+      rules: [
+        {
+          name: 'noProxyFee',
```
Comment (Asmadek) [inline src/renderer/widgets/Unstake/model/form-model.ts:91]: do we need error text here?

File: src/renderer/widgets/Unstake/model/form-model.ts [hunk 3]
```diff
@@ -0,0 +1,690 @@
...
+          }),
+          validator: (_s, _f, { isProxy, proxyBalance, fee }) => {
+            if (!isProxy) return true;
+
+            return new BN(fee).lte(new BN(proxyBalance));
+          },
+        },
+        {
+          name: 'noUnstakeBalance',
+          errorText: 'No unstake balance',
```
Comment (Asmadek) [inline src/renderer/widgets/Unstake/model/form-model.ts]: Use i18n here

File: src/renderer/widgets/Unstake/model/form-model.ts [hunk 4]
```diff
@@ -0,0 +1,690 @@
...
+        },
+      ],
+    },
+    description: {
+      init: '',
+      rules: [
+        {
+          name: 'maxLength',
+          errorText: 'transfer.descriptionLengthError',
+          validator: (value) => !value || value.length <= 120,
```
Comment (Asmadek) [inline src/renderer/widgets/Unstake/model/form-model.ts]: I still think that we need constant for max description length

File: src/renderer/widgets/Unstake/model/form-model.ts [hunk 5]
```diff
@@ -0,0 +1,690 @@
...
+
+sample({
+  source: {
+    staking: $staking,
+    networkStore: $networkStore,
+    shards: $unstakeForm.fields.shards.$value,
+  },
+  filter: ({ staking, networkStore }) => Boolean(staking) && Boolean(networkStore),
+  fn: ({ staking, networkStore, shards }) => {
+    if (shards.length === 0) return '0';
```
Comment (Asmadek) [inline src/renderer/widgets/Unstake/model/form-model.ts]: We have `'0'` three times in the function and some in other places, maybe add constant for it?

**Review by Asmadek — APPROVED** [review]

**Pull request merged**